app/Http/Controllers/Frontend/Auth/AuthController.php

```diff
@@ -80,4 +84,54 @@ protected function create(array $data)
                 'password' => bcrypt($data['password']),
         ]);
     }
+
+    /**
+     * Redirect the user to the Facebook authentication page.
+     *
+     * @return Response
+     */
+    public function redirectToProvider()
+    {
+        return Socialite::driver('facebook')->redirect();
+    }
+
+    /**
+     * Obtain the user information from Facebook.
+     *
+     * @return Response
+     */
+    public function handleProviderCallback()
+    {
+        try {
+            $user = Socialite::driver('facebook')->user();
+        } catch (Exception $e) {
+            return redirect('auth/facebook');
+        }
+
+        $authUser = $this->findOrCreateUser($user);
+
+        Auth::login($authUser, true);
+
+        return redirect()->route('getContact');
+    }
+
+    /**
+     * Return user if exists; create and return if doesn't
+     *
+     * @param int $facebookUser facebookUser
+     *
+     * @return loginUser
+     */
+    private function findOrCreateUser($facebookUser)
+    {
+        $authUser = User::where('facebook_id', $facebookUser->id)->first();
```

Comment:
if you use `firstOrNew` function then you don't need this query. please check the `firstOrNew` document.